Add Jupyter Book documentation infrastructure with auto-generated API reference via sphinx-autoapi (#272)
mllam/neural-lam: docs/getting-started/quickstart.md

# Quickstart

## Overview

This guide walks through the minimum steps to get started with Neural-LAM:
1. Set up example data
2. Create a graph
3. Train a model
4. Evaluate the model

## Step 1: Download Example Data

When running tests for the first time, example data is automatically downloaded from S3. You can trigger this download by running a minimal test.

```{code-block} bash
# Run a minimal test to trigger the data download
pytest tests/test_training.py -vv -s -k "test_training"
```

## Step 2: Create a Graph

Before training, you must construct a graph mesh for your data.

```{code-block} bash
python -m neural_lam.create_graph --config_path <path-to-config> --name <graph-name>
```

This script builds the mesh graph required by the GNN models.

## Step 3: Train a Model

Now you can train a model using the graph and configuration.

```{code-block} bash
python -m neural_lam.train_model --config_path <path-to-config> --model graph_lam --graph <graph-name>
```

> Neural-LAM supports several models like `graph_lam`, `hilam`, and `hi_lam_parallel`.

Reviewer: `hilam` is not a valid key, `MODELS` in `neural_lam/models/__init__.py` has `graph_lam` / `hi_lam` / `hi_lam_parallel`, so `--model hilam` fails on `choices`. Same point as last round, just didn't land.

```suggestion
Neural-LAM supports several models like `graph_lam`, `hi_lam`, and `hi_lam_parallel`.
```

# Quickstart

## Overview

This guide walks through the minimum steps to get started with Neural-LAM:
1. Set up example data
2. Create a graph
3. Train a model
4. Evaluate the model

## Step 1: Download Example Data

When running tests for the first time, example data is automatically downloaded from S3. You can trigger this download by running a minimal test.

```{code-block} bash
# Run a minimal test to trigger data download

> pytest tests/test_training.py -vv -s -k "test_training" --co

Reviewer: `--co` is collect-only, it never runs the fixture, so it won't download anything. Drop it.

```suggestion
# Run a minimal test to trigger the data download
pytest tests/test_training.py -vv -s -k "test_training"
```

```

> Alternatively, you can use the `DummyDatastore` for quick testing without downloading real data.

Reviewer: `DummyDatastore` is a test fixture (`tests/dummy_datastore.py`), not selectable through the config/CLI flow this guide describes. I would drop it.

```suggestion
```

## Step 2: Create a Graph

Before training, you must construct a graph mesh for your data.

```{code-block} bash
python -m neural_lam.create_graph --config_path <path-to-config> --name <graph-name>
```

This script builds the mesh graph required by the GNN models.

## Step 3: Train a Model

Now you can train a model using the graph and configuration.

```{code-block} bash
python -m neural_lam.train_model --config_path <path-to-config> --model graph_lam --graph <graph-name>
```

> Neural-LAM supports several models like `graph_lam`, `hilam`, and `hilam_parallel`.

Reviewer: keys are `hi_lam` / `hi_lam_parallel` (see `MODELS` in `neural_lam/models/__init__.py`), so `--model hilam` fails on `choices`.

```suggestion
Neural-LAM supports several models like `graph_lam`, `hi_lam`, and `hi_lam_parallel`.
```
